### ISSUE TITLE
pipewire sometimes crashes on Fedora 35 with this plugin

### ISSUE THREAD

**Issue by ifoolb** [werman/noise-suppression-for-voice#116]
I followed this article [Noise Cancellation on Linux](https://medium.com/@gamunu/linux-noise-cancellation-b9f997f6764d) to setup the library for pipewire on Fedora 35. But it sometimes crashes, usually when an application uses the microphone like Microsoft teams. And it won't be able to use the config unless I run: ```systemctl --user restart pipewire``` , otherwise, pipewire crashes with the following error:

```
[user@fedora ~]$ pipewire -c /home/avatar/.config/pipewire/input-filter-chain.conf 
/usr/include/c++/11/bits/stl_vector.h:1045: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = float; _Alloc = std::allocator<float>; std::vector<_Tp, _Alloc>::reference = float&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__n < this->size()' failed.
Aborted (core dumped)
```

**Comment by athre0z** [comment]
I also encounter this on Arch. Stack trace:

```
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007f3027c8e3d3 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007f3027c3e838 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007f3027c28535 in __GI_abort () at abort.c:79
#4  0x00007f3026ed2002 in std::__glibcxx_assert_fail (file=<optimized out>, line=<optimized out>, function=<optimized out>, condition=<optimized out>) at /usr/src/debug/gcc/libstdc++-v3/src/c++11/debug.cc:60
#5  0x00007f302752e4a6 in RnNoiseCommonPlugin::process(float const*, float*, int, float) () from /usr/lib64/ladspa/librnnoise_ladspa.so
#6  0x00007f302752e6e6 in ladspa::builder<RnNoiseStereo>::_run(void*, unsigned long) () from /usr/lib64/ladspa/librnnoise_ladspa.so
#7  0x00007f302754f40f in capture_process (d=0x555bcdb62b90) at ../pipewire/src/modules/module-filter-chain.c:632
#8  0x00007f302802393b in impl_node_process_input (object=0x555bcdb96f80) at ../pipewire/src/pipewire/stream.c:991
#9  0x00007f302717cd91 in impl_node_process (object=0x7f3027125038) at ../pipewire/spa/plugins/audioconvert/audioadapter.c:1390
#10 0x00007f3028002289 in process_node (data=0x555bcdba4270) at ../pipewire/src/pipewire/impl-node.c:1090
#11 0x00007f302800592c in node_on_fd_events (source=<optimized out>) at ../pipewire/src/pipewire/impl-node.c:1148
#12 0x00007f3028092e4d in loop_iterate (object=<optimized out>, timeout=<optimized out>) at ../pipewire/spa/plugins/support/loop.c:430
#13 0x00007f3027fe2d37 in do_loop (user_data=0x555bcdb25ed0) at ../pipewire/src/pipewire/data-loop.c:81
#14 0x00007f3027c8c54d in start_thread (arg=<optimized out>) at pthread_create.c:442
#15 0x00007f3027d11874 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:100
```

**Comment by athre0z** [comment]
Debugged into this and it turns out it's a duplicate of https://github.com/werman/noise-suppression-for-voice/issues/79. Because the last official release was 0.91 from 2020, the fix never made it into distro repos. It'd be great if you could consider creating a new release @werman.

**Comment by werman** [comment]
@athre0z I'm planning to make a big release soon which also has this issue fixed.

**Comment by werman** [comment]
Released v1.01, should be fixed now.
